src/main/java/the_warlord/powers/ImprovisePower.java

```diff
@@ -0,0 +1,60 @@
+package the_warlord.powers;
+
+import com.megacrit.cardcrawl.actions.utility.UseCardAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.cards.CardQueueItem;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.monsters.AbstractMonster;
+import com.megacrit.cardcrawl.powers.AbstractPower;
+import the_warlord.util.CardMetaUtils;
+
+public class ImprovisePower extends CustomWarlordModPower {
+    public static final StaticPowerInfo STATIC = StaticPowerInfo.Load(ImprovisePower.class);
+    public static final String POWER_ID = STATIC.ID;
+    private int cardsPlayed = 0;
+
+    public ImprovisePower(AbstractCreature owner, int amount) {
+        super(STATIC);
+        this.type = PowerType.BUFF;
+        this.owner = owner;
+        this.amount = amount;
+        loadRegion("flight");
+
+        updateDescription();
+    }
+
+    @Override
+    public void updateDescription() {
+        this.description = String.format(DESCRIPTIONS[0], this.amount);
+    }
+
+    @Override
+    public void atStartOfTurnPostDraw() {
+        if (!this.owner.hasPower(PosturePower.POWER_ID)) {
+            flash();
+            this.cardsPlayed = 0;
+        }
+    }
+
+    public void onUseCard(AbstractCard card, UseCardAction action) {
+        if (!card.purgeOnUse && this.amount > 0 && AbstractDungeon.actionManager.cardsPlayedThisTurn.size() - this.cardsPlayed <= 1 && (!this.owner.hasPower(PosturePower.POWER_ID))) {
```

Comment:
Your current implementation doubles the first card played once for each stack of ImprovisePower, if I'm not mistaken. 
Echo Form, on the other hand, plays the first X cards twice, where X is the number of Echo Form stacks. 
I think we should stick to how Echo Form stacks for this card, since it has almost the same text. 

That would mean that this if-statement should check if the number of cards played this turn is <= this.amount.